packages/core-flows/src/product/workflows/update-product-variants.ts

```diff
@@ -15,6 +23,72 @@ export const updateProductVariantsWorkflow = createWorkflow(
   (
     input: WorkflowData<WorkflowInput>
   ): WorkflowData<ProductTypes.ProductVariantDTO[]> => {
-    return updateProductVariantsStep(input)
+    // Passing prices to the product module will fail, we want to keep them for after the variant is updated.
+    const updateWithoutPrices = transform({ input }, (data) => {
+      return {
+        selector: data.input.selector,
+        update: {
+          ...data.input.update,
+          prices: undefined,
+        },
+      }
+    })
+
+    const updatedVariants = updateProductVariantsStep(updateWithoutPrices)
+
+    // We don't want to do any pricing updates if the prices didn't change
+    const variantIds = transform({ input, updatedVariants }, (data) => {
+      if (!data.input.update.prices) {
+        return []
+      }
+
+      return data.updatedVariants.map((v) => v.id)
+    })
+
+    const variantPriceSetLinks = getVariantPricingLinkStep({
+      ids: variantIds,
+    })
+
+    const pricesToUpdate = transform(
+      { input, variantPriceSetLinks },
+      (data) => {
+        if (!data.variantPriceSetLinks.length) {
+          return {}
+        }
+
+        return {
+          selector: {
+            ids: data.variantPriceSetLinks.map((link) => link.price_set_id),
+          } as PricingTypes.FilterablePriceSetProps,
+          update: {
+            prices: data.input.update.prices,
+          } as PricingTypes.UpdatePriceSetDTO,
+        }
+      }
+    )
+
+    const updatedPriceSets = updatePriceSetsStep(pricesToUpdate)
+
+    // We want to correctly return the variants with their associated price sets and the prices coming from it
+    return transform(
+      {
+        variantPriceSetLinks,
+        updatedVariants,
+        updatedPriceSets,
+      },
+      (data) => {
+        return data.updatedVariants.map((variant, i) => {
+          const linkForVariant = data.variantPriceSetLinks.find(
+            (link) => link.variant_id === variant.id
+          )
+
+          const priceSetForVariant = data.updatedPriceSets.find(
+            (priceSet) => priceSet.id === linkForVariant?.price_set_id
+          )
```

Comment:
nit: convert to maps, this is one of those potentially heavy operations

Reply:
@olivermrbl I'd like to do some benchmarking and optimization as a separate task if you agree (I'm taking notes on this)